libgalois/include/katana/ParallelSTL.h

```diff
@@ -391,6 +393,103 @@ partial_sum(InputIt first, InputIt last, OutputIt d_first) {
   }
 }
 
+template <class InputIt, class OutputIt, class UnaryOperation>
+OutputIt
+transform(
+    InputIt first, InputIt last, OutputIt d_first, UnaryOperation unary_op) {
+  // This will only work for random access iterators
+  using input_category =
+      typename std::iterator_traits<InputIt>::iterator_category;
+  using output_category =
+      typename std::iterator_traits<OutputIt>::iterator_category;
+  static_assert(
+      std::is_base_of_v<std::random_access_iterator_tag, input_category>);
+  static_assert(
+      std::is_base_of_v<std::random_access_iterator_tag, output_category>);
+
+  using diff_type = typename std::iterator_traits<InputIt>::difference_type;
+
+  diff_type length = std::distance(first, last);
+  katana::do_all(
+      katana::iterate(diff_type{0}, length),
+      [&](diff_type index) { *(d_first + index) = unary_op(*(first + index)); },
+      katana::steal());
```

Comment:
Don't use `steal()` for _operators_ that are not imbalanced. Typically, imbalanced operators iterate over edges (which could be imbalanced).